### ISSUE TITLE
Get back into Stackage Nightly with GHC 9.8

### ISSUE THREAD

**Issue by ysangkok** [dhall-lang/dhall-haskell#2563]
dhall packages were thrown out because of bound issues:

See [relevant section of build-constraints.yml](https://github.com/commercialhaskell/stackage/blob/ff36aca21a6987c1203b69349d905a44e46da216/build-constraints.yaml#L6664C26-L6664C26)

**Comment by ysangkok** [comment]
Fixed by #2584

**Comment by krakrjak** [comment]
Bump... Seems it Dhall didn't make the cut for stackage 23.0. I would love to see it make a triumphant return in 23.1 if possible.

**Comment by TeofilC** [comment]
Hopefully https://github.com/dhall-lang/dhall-haskell/pull/2602 should have relaxed all the necessary bounds so maybe this is just waiting on a Hackage revision/release?

**Comment by ysangkok** [comment]
@TeofilC Unfortunately not, as lts-23.0 has Diff-0.5 which is not supported by the revised dhall-1.42.1:

`cabal get dhall`, `cd dhall-1.42.1`, `stack init --resolver lts-23.0 .` reports:
```
             Diff version 0.5 found
                 - dhall requires >=0.2 && <0.5
```

Note that the newest version is Diff-1.0.2 which is in Nightly already.

Another issue is `special-values`, which doesn't support bytestring-0.12.

* https://github.com/minad/special-values/issues/5

Doctests run with `cabal repl --with-ghc=doctest` also fail. But maybe that's just not the right way to run them? I like that method because it means avoiding environment files and statefulness. Here is an example error:

```
src/Dhall/Deriving.hs:307: failure in expression `input person "./simon.dhall"'
expected: Person {personName = Name {getName = \"Simon\"}, personFavoriteFont = ComicSans}
 but got: *** Exception:
          ^
          \8627 ./simon.dhall
          
          \ESC[1;31mError\ESC[0m: Missing file ./simon.dhall
          
          1\9474 ./simon.dhall
          
          (input):1:1
```

**Comment by ysangkok** [comment]
@Gabriella439 There has been a few source changes since 1.42.1, which is the last release available on Hackage. Would it be possible to make a new upload? That would address the compatibility with GHC 9.8, since TeofilC has fixed the issue on the default branch.

**Comment by Gabriella439** [comment]
Yeah, sorry about the delay.  I will push out a new release soon

**Comment by Gabriella439** [comment]
I've begun the process of cutting a language standard release: https://github.com/dhall-lang/dhall-lang/pull/1393

Once that merges I'll cut a release of the Haskell implementation

**Comment by ysangkok** [comment]
Closing this because the main package was added to the LTS for GHC 9.8:

https://www.stackage.org/diff/lts-23.8/lts-23.9

Full support is tracked in #2638

**Comment by krakrjak** [comment]
Thank ya'll for the hard work getting this done. It's very much appreciated.
